keyboards/tapedeck/tapedeck.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2023 Gowla
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#ifdef POT_ENABLE
+  #include "analog.h"
+#endif
+
+int16_t pot_val = 0;
+int16_t prev_val = 0;
+int16_t val = 0;
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) {
+        return false;
+    }
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_WH_U);
+        } else {
+            tap_code(KC_WH_D);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_BRIU);
+        } else {
+            tap_code(KC_BRID);
+        }
+    }
+    return true;
+}
+
+void keyboard_post_init_kb(void) {
+#ifdef POT_ENABLE
+    analogReference(ADC_REF_POWER);
+#endif
+    keyboard_post_init_user();
+}
+
+uint8_t divisor = 0;
+
+long map(long x, long in_min, long in_max, long out_min, long out_max)
+{
+  return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+}
+
+void slider(void) {
+    if (divisor++) { // only run the slider function 1/256 times it's called
+        return;
+    }
+
+    pot_val = analogReadPin(D4);
+	val = map(pot_val, 0, 1023, 1, 50);
+
+	if (( val > (prev_val + 1)) && val != prev_val){
+		int i;
+		for (i = prev_val; i < val; i++ )
+				{tap_code(KC_VOLU);}
+	}
+	if ((val  < (prev_val - 1)) && val != prev_val){
+		int i;
+		for (i = val; i < prev_val; i++ )
+				{tap_code(KC_VOLD);}
+	}
+	prev_val = val;
+}
```

Comment:
Actually it won't fail, because the functions are not `static`, so the compiler won't be able to detect that they are unused.  Does not mean that it's a good style though.

Reply:
I got build failures, but looks like i didnt digest them correctly.

```
Compiling: keyboards/tapedeck/tapedeck.c                                                           Compiling: .build/obj_tapedeck/src/default_keyboard.c                                              Compiling: quantum/led.c                                                                           Compiling: quantum/quantum.c                                                                       Compiling: quantum/bitwise.c                                                                       Compiling: quantum/keymap_introspection.c                                                          keyboards/tapedeck/tapedeck.c: In function 'slider':
keyboards/tapedeck/tapedeck.c:66:15: error: implicit declaration of function 'analogReadPin' [-Werror=implicit-function-declaration]
   66 |     pot_val = analogReadPin(D4);
      |               ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
 [OK]
```

Reply:
Thank you for looking, intention is allow user to disable slider by commenting out the define. Would you like this section wrapped with #ifdef POT_ENABLE ? It does not get called in the housekeeping task below if not defined.